tf-psa-crypto/include/psa/crypto.h

```diff
@@ -5768,6 +5768,49 @@ psa_status_t psa_export_public_key_iop_complete(psa_export_public_key_iop_t *ope
                                                 size_t data_size,
                                                 size_t *data_length);
 
+/**
+ * \brief                       Abort an interruptible public-key export operation.
+ *
+ * \warning                     This is a beta API, and thus subject to change
+ *                              at any point. It is not bound by the usual
+ *                              interface stability promises.
+ *
+ * \note                        This function clears the number of ops completed
+ *                              as part of the operation. Please ensure you copy
+ *                              this value via
+ *                              \c psa_export_public_key_iop_get_num_ops() if
+ *                              required before calling.
+ *
+ * \note                        Aborting an operation frees all
+ *                              associated resources except for the operation
+ *                              structure itself. Once aborted, the operation
+ *                              object can be reused for another operation by
+ *                              calling \c psa_export_public_key_iop_setup() again.
+ *
+ * \note                        You may call this function any time after the
+ *                              operation object has been initialized.
+ *                              In particular, calling
+ *                              \c psa_export_public_key_iop_abort() after the
+ *                              operation has already been terminated by a call
+ *                              to \c psa_export_public_key_iop_abort() or
+ *                              \c psa_export_public_key_iop_complete() is safe.
+ *
+ * \param[in,out] operation     The \c psa_export_public_key_iop_t to use
+ *
+ * \retval #PSA_SUCCESS
+ *          The operation was aborted successfully.
+ *
+ * \retval #PSA_ERROR_NOT_SUPPORTED \emptydescription
```

Comment:
Which version of the API did you use? Just asking, because I am looking at https://github.com/user-attachments/files/15927984/IHI0086-PSA_Certified_Crypto_API-1.3.0-interruptible-operations-draft3.pdf and there we don't return `PSA_ERROR_NOT_SUPPORTED` but we can return `PSA_ERROR_COMMUNICATION_FAILURE`.

Reply:
`NOT_SUPPORTED` doesn't make sense to me in an abort function. It doesn't take any non-opaque parameters, so there's nothing to report as not supported.

`COMMUNICATION_FAILURE` can't be raised inside our code base. It's meant for client-server applications, when a client stub can't reach the server. So it's an error that can users of Mbed TLS can encounter, but only if they build with `MBEDTLS_PSA_CRYPTO_CLIENT && !MBEDTLS_PSA_CRYPTO_C`. This is true for every function except for the attribute accessors. I don't think we've been consistent about including it in function documentation, so it doesn't matter whether we include it here or not.